java/src/main/java/com/google/cloud/dataproc/templates/gcs/GCSToJDBC.java

```diff
@@ -47,13 +54,39 @@ public static GCSToJDBC of(String... args) {
   public void runTemplate() {
     validateInput();
     try (SparkSession spark = SparkSession.builder().appName("GCS to JDBC").getOrCreate()) {
-      Dataset<Row> dataset =
-          spark.read().format(config.getInputFormat()).load(config.getInputLocation());
-
+      Dataset<Row> dataset = load(spark);
       write(dataset);
     }
   }
 
+  public Dataset<Row> load(SparkSession spark) {
+    Dataset<Row> inputData = null;
+
+    switch (config.getInputFormat()) {
+      case GCS_JDBC_CSV_FORMAT:
+        inputData =
+            spark
+                .read()
+                .format(config.getInputFormat())
+                .option(GCS_JDBC_CSV_HEADER, true)
+                .option(GCS_JDBC_CSV_INFER_SCHEMA, true)
+                .load(config.getInputLocation());
+        break;
+      case GCS_JDBC_AVRO_FORMAT:
+      case GCS_JDBC_ORC_FORMAT:
+        inputData = spark.read().format(config.getInputFormat()).load(config.getInputLocation());
+        break;
+      case GCS_JDBC_PRQT_FORMAT:
+        inputData = spark.read().parquet(config.getInputLocation());
```

Comment:
I would prefer if we can use same code as for Avro and ORC. 
Why are we using a different code for parquet ?
`inputData = spark.read().format(config.getInputFormat()).load(config.getInputLocation());`

Reply:
I followed what had been done previously in the `GCSToBigQuery.java` template. That one has hardcoded options for `inferSchema` and `header`. So does the equivalent Python `gcs_to_jsbc.py` template.